src/Symfony/Component/HttpFoundation/Request.php

```diff
@@ -1203,11 +1203,13 @@ public function getMethod(): string
 
         $method = strtoupper($method);
 
-        if (\in_array($method, ['GET', 'HEAD', 'POST', 'PUT', 'DELETE', 'CONNECT', 'OPTIONS', 'PATCH', 'PURGE', 'TRACE', 'QUERY'], true)) {
+        if (\in_array($method, ['GET', 'HEAD', 'CONNECT', 'TRACE'], true)) {
+            trigger_deprecation('symfony/http-foundation', '7.4', 'HTTP method override is deprecated for methods GET, HEAD, CONNECT and TRACE; it will be ignored in Symfony 8.0.', $method);
+
             return $this->method = $method;
         }
 
-        if (!preg_match('/^[A-Z]++$/D', $method)) {
+        if (\strlen($method) !== strspn($method, 'ABCDEFGHIJKLMNOPQRSTUVWXYZ')) {
```

Comment:
```suggestion
        if (\strlen($method) !== \strspn($method, 'ABCDEFGHIJKLMNOPQRSTUVWXYZ')) {
```

Reply:
we use the `\` prefix only for function calls that are optimized as opcodes by opcache ;)